jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -0,0 +1,121 @@
+            [zoom_y_min, zoom_y_max, zoom_y_max, zoom_y_min])
+
+        # radius for querying the region is based on:
+        # the distance between the longest zoom limits and the center point
+        zoom_radius = max(skycoord_center.separation(zoom_coordinate))
+
+        # conducts search based on SDSS
+        if self.catalog_selected == "SDSS":
+            # queries the region (based on the provided center point and radius)
+            # finds all the sources in that region
+            query_region_result = SDSS.query_region(skycoord_center, radius=zoom_radius,
+                                                    data_release=17)
+            self.results_available = True
+
+        else:
+            self.results_available = False
+            self.number_of_results = 0
+            raise ValueError(f"{self.catalog_selected} not a supported catalog")
+
+        # nothing happens in the case the query returned empty
+        if query_region_result is None:
+            self.number_of_results = 0
+            return
+
+        # a table is created storing the 'ra' and 'dec' plottable points of each source found
+        skycoord_table = SkyCoord(query_region_result['ra'], query_region_result['dec'], unit='deg')
+
+        # coordinates found are converted to pixel coordinates
+        pixel_table = viewer.state.reference_data.coords.world_to_pixel(skycoord_table)
+        # coordinates are filtered out (using a mask) if outside the zoom range
+        pair_pixel_table = np.dstack((pixel_table[0], pixel_table[1]))
+        # ma.masked_outside removes the coordinates outside the zoom range
+        # ma.compress_rows removes any row that has a mask mark
+        filtered_table = ma.compress_rows(
+            ma.masked_outside(pair_pixel_table, [zoom_x_min, zoom_y_min], [zoom_x_max, zoom_y_max])
+            [0])
+        # coordinates are split into their respective x and y values
+        # then they are converted to sky coordinates
+        filtered_pair_pixel_table = np.array(np.hsplit(filtered_table, 2))
+        x_coordinates = np.squeeze(filtered_pair_pixel_table[0])
+        y_coordinates = np.squeeze(filtered_pair_pixel_table[1])
+        filtered_skycoord_table = viewer.state.reference_data.coords.pixel_to_world(x_coordinates,
+                                                                                    y_coordinates)
+
+        # QTable stores all the filtered sky coordinate points to be marked
+        catalog_results = QTable({'coord': filtered_skycoord_table})
+        self.number_of_results = len(catalog_results)
+
+        # markers are added to the viewer based on the table
+        viewer.marker = {'color': 'red', 'alpha': 0.8, 'markersize': 5, 'fill': False}
+        viewer.add_markers(table=catalog_results, use_skycoord=True, marker_name='catalog_results')
+
+    def vue_do_search(self, *args, **kwargs):
+        # calls self.search() which handles all of the searching logic
+        self.search()
```

Comment:
What was the motivation to do this instead of having the actual logic in this method?

Reply:
I think this was from my suggestion to make the user-API more friendly (instead of blinking accepting args/kwargs to something exposed to the user).  But if you think we should defer that to a proper user-friendly-plugin-API effort, I'm ok with reverting.

Reply:
If the effort is happening soon, I guess we can keep it as-is. I'll defer to Kyle's judgement here. Thanks for the clarification!